packages/app/src/components/popup/container.tsx

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+import {
+  Box,
+  Flex,
+  Text,
+  ArrowIcon,
+  BoxProps,
+  FlexProps,
+  StxNexus,
+  IconButton,
+  color,
+} from '@stacks/ui';
+import styled from '@emotion/styled';
+import { SettingsPopover } from './settings-popover';
+import { useDrawers } from '@common/hooks/use-drawers';
+import { useAnalytics } from '@common/hooks/use-analytics';
+import { ScreenPaths } from '@store/onboarding/types';
+import { IconMenu2, IconFlask } from '@tabler/icons';
+import { useRecoilValue } from 'recoil';
+import { currentNetworkStore } from '@store/recoil/networks';
+import { hasRehydratedVaultStore } from '@store/recoil/wallet';
+import { ChainID } from '@stacks/transactions';
+
+const CloseIconContainer = styled(Box)`
+  svg {
+    height: 14px;
+    opacity: 50%;
+    // margin-top: 10px;
+  }
+`;
+
+const Header: React.FC<FlexProps> = props => {
+  const { doChangeScreen } = useAnalytics();
+
+  return (
+    <Flex
+      cursor="pointer"
+      flexDirection="row"
+      onClick={() => doChangeScreen(ScreenPaths.HOME)}
+      {...props}
+    >
+      <Box>
+        <StxNexus
+          height="23px"
+          position="relative"
+          top="7px"
+          color="black"
+          display="inline-block"
+        />
+      </Box>
+      <Box>
+        <Text
+          fontFamily="heading"
+          color="ink.1000"
+          fontWeight="600"
+          ml="base-tight"
+          // fontSize={[2, 4]}
+          fontSize={4}
+        >
+          Stacks Wallet
+        </Text>
+      </Box>
+    </Flex>
+  );
+};
+
+const ModeBadge: React.FC<FlexProps> = props => {
+  const { chainId } = useRecoilValue(currentNetworkStore);
+  const mode = chainId === ChainID.Mainnet ? 'Mainnet' : 'Testnet';
```

Comment:
Recommend a wrapper fn with signature: `whenNetwork({ mainnet: 'Mainnet', testnet: 'Testnet' })`

Could better handle a, improbable, but nasty situation where chainid is neither.